src/OperationResponse.php

```diff
@@ -495,4 +514,10 @@ private function hasProtoResponse()
     {
         return !is_null($this->lastProtoResponse);
     }
+
+    private function isNewSurfaceOperationsClient(): bool
+    {
+        return !$this->operationsClient instanceof LegacyOperationsClient
+            && false !== strpos(get_class($this->operationsClient), '\\Client\\');
```

Comment:
Nit: maybe make the magic value `'\\Client\\'` into a private constant `NEW_CLIENT_NAMESPACE` for readability 🤷🏻‍♂️

Reply:
Good point